content/frontend/react-apollo/9-pagination.md

```diff
@@ -102,261 +158,349 @@ export const FEED_QUERY = gql`
+retrieve the variables. Let's implement the
+`getQueryVariables` function.
 
-```js(path=".../hackernews-react-apollo/src/components/LinkList.js")
-<Query query={FEED_QUERY} variables={this._getQueryVariables()}>
+</Instruction>
+
+```js{}(path=".../hackernews-react-apollo/src/components/LinkList.js")
+import { LINKS_PER_PAGE } from '../constants';
+
+// ...
+
+const getQueryVariables = (isNewPage, page) => {
+  const skip = isNewPage ? (page - 1) * LINKS_PER_PAGE : 0;
+  const take = isNewPage ? LINKS_PER_PAGE : 100;
+  const orderBy = { createdAt: 'desc' };
+  return { take, skip, orderBy };
+};
 ```
 
+<Instruction>
+
+The `getQueryVariables` function is responsible for
+returning values for `skip`, `take`, and `orderBy`. For
+`skip`, we first check whether we are currently on the
+`/new` route. If so, the value for `skip` is the current
+page (subtracting `1` to handle the index) multiplied by the
+`LINKS_PER_PAGE` contstant. If we're not on the `/new`
+route, the value for `skip` is `0`. We use the same
+`LINKS_PER_PAGE` constant to determine how many links to
+`take`.
+
 </Instruction>
 
-You're now passing `first, skip, orderBy` values as `variables` based on the current page (`this.props.match.params.page`) which it's used to calculate the chunk of links that you retrieve.
+We're now passing `take, skip, orderBy` values as
+`variables` based on the current page.
 
-Also note that you're including the ordering attribute `createdAt_DESC` for the `new` page to make sure the newest links are displayed first. The ordering for the `/top` route will be calculated manually based on the number of votes for each link.
+Also note that we're including the ordering attribute
+`{ createdAt: 'desc' }` for the `new` page to make sure the
+newest links are displayed first. The ordering for the
+`/top` route will be calculated manually based on the number
+of votes for each link.
 
-You also need to define the `LINKS_PER_PAGE` constant and then import it into the `LinkList` component.
+We also need to define the `LINKS_PER_PAGE` constant and
+then import it into the `LinkList` component.
 
 <Instruction>
 
 Open `src/constants.js` and add the following definition:
 
 ```js(path=".../hackernews-react-apollo/src/constants.js")
-export const LINKS_PER_PAGE = 5
+export const LINKS_PER_PAGE = 5;
 ```
 
 </Instruction>
 
-### Implementing navigation
+### Implementing Navigation
 
-Next, you need functionality for the user to switch between the pages. First add two `button` elements to the bottom of the `LinkList` component that can be used to navigate back and forth.
+Next, we need functionality for the user to switch between
+the pages. First add two `button` elements to the bottom of
+the `LinkList` component that can be used to navigate back
+and forth.
 
 <Instruction>
 
 Open `LinkList.js` and update `render` to look as follows:
 
 ```js{11-15,23,27-36}(path=".../hackernews-react-apollo/src/components/LinkList.js")
```

Comment:
Seems the contents of the JSX in `LinkedList.js` changed. I think it would be best to remove the diff here 🙂
```suggestion
```js(path=".../hackernews-react-apollo/src/components/LinkList.js")
```